lldb/test/API/commands/frame/recognizer/categories

```diff
@@ -1 +0,0 @@
-objc
```

Comment:
Hmm are we sure we aren't losing test coverage by converting this to plain C? I don't think it'd be a huge deal if we just added an entirely new test as part of this PR

Reply:
Maybe with some Makefile trickery you could add a separate main.c in this directory (in addition to `main.m`). But that's probably not worth the trouble

Reply:
The intended test coverage of this test case seems to be to test the frame recognizers as indicated by the file path.
Making making it run also on non-Mac platforms, we are hence gaining test coverage on other systems.

There is no comment I could find which states that frame recognizers work differently for Objective-C. As such, I would expect any test coverage for Objective-C to be incidental, and would expect the Objective-C support to be tested in-depth by some other test cases.

I am used to the guideline "It's ok to remove incidental test coverage. If the test coverage was worthwhile, there should have been a comment added as part of adding the test" from other projects. But that's based on my experience with how other projects treat their test suites. Not sure if this also applies to LLDB?

Reply:
Yea that's fair, there doesn't seem to be anything objc specific here, so seems ok to turn it into C and enable it on all platforms.

The original change that introduced these did mention Objective-C as one of the motivators: https://reviews.llvm.org/D44603. And there seems to have been changes made to `AppleObjCStackFrameRecognizer.cpp` in early version of the revision (though Phabricator doesn't allow me to view those).

So overall I'm ok with this

Reply:
The early motivation for stack frame recognizers was to get the "current exception object" from the objc_exception_throw routine.  But this patch didn't add that recognizer, and when it was added it had its own test.  There's nothing about this test that's ObjC specific, and the ObjC specific bits that motivated making the test file a .m file were actually tested in a separate commit.  So I'm also fine with changing this test to a .c.

Reply:
Thanks for confirming!